common/json.c

```diff
@@ -412,8 +412,17 @@ void json_add_literal(struct json_result *result, const char *fieldname,
 
 void json_add_string(struct json_result *result, const char *fieldname, const char *value)
 {
+	char *escaped = tal_arr(result, char, strlen(value) * 2 + 1);
+	size_t i, n;
+
 	json_start_member(result, fieldname);
-	result_append_fmt(result, "\"%s\"", value);
+	for (i = n = 0; value[i]; i++) {
+		if (value[i] == '\\' || value[i] == '"')
```

Comment:
We might want to escape a lot more than that (`\r`, `\n`, `\0`, ...). I'd suggest replacing all non-printable characters with a space or a '.'.

Reply:
Yes, I looked at the json spec and there are more to escape.  I'll do it properly.